File: .tmp_py4013/batch_04_sources/2506_count_pairs_of_similar_strings.js
// LeetCode 2506 - Count Pairs Of Similar Strings
// https://leetcode.com/problems/count-pairs-of-similar-strings/

/**
 * @param {string[]} words
 * @return {number}
 */
var similarPairs = function(words) {
    const freq = new Map();
    let ans = 0;
    for (const w of words) {
        let mask = 0;
        for (let i = 0; i < w.length; i++) mask |= 1 << (w.charCodeAt(i) - 97);
        ans += freq.get(mask) || 0;
        freq.set(mask, (freq.get(mask) || 0) + 1);
    }
    return ans;
};

File: .tmp_rs4013/rust_check_batch08/2506_count_pairs_of_similar_strings.rs
struct Solution;
// LeetCode 2506 - Count Pairs Of Similar Strings
// https://leetcode.com/problems/count-pairs-of-similar-strings/

use std::collections::HashMap;

impl Solution {
    pub fn similar_pairs(words: Vec<String>) -> i32 {
        let mut freq: HashMap<u32, i32> = HashMap::new();
        let mut ans = 0;
        for w in words {
            let mut mask = 0u32;
            for c in w.bytes() {
                mask |= 1 << (c - b'a');
            }
            ans += *freq.get(&mask).unwrap_or(&0);
            *freq.entry(mask).or_insert(0) += 1;
        }
        ans
    }
}

fn main() {}
